### ISSUE TITLE
Unable to connect to HTTP

### ISSUE THREAD

**Issue by wicaksono** [Microsoft/WSL#3277]
This bug-tracker is monitored by developers and other technical types.  We like detail!  So please use this form and tell us, concisely but precisely, what's up.  Please fill out ALL THE FIELDS!

If you have a feature request, please post to [the UserVoice](https://wpdev.uservoice.com/forums/266908).

If this is a console issue (a problem with layout, rendering, colors, etc.), please post to the [console issue tracker](https://github.com/microsoft/console/issues).

**Important: When reporting BSODs or security issues, DO NOT attach memory dumps, logs, or traces to Github issues**.  Instead, send dumps/traces to secure@microsoft.com, referencing the GitHub bug number. Ideally, please [configure your machine to capture minidumps](https://support.microsoft.com/en-us/help/315263/how-to-read-the-small-memory-dump-file-that-is-created-by-windows-if-a), repro the issue, and send the minidump from "C:\Windows\minidump\".

* Your Windows build number:  Microsoft Windows [Version 10.0.17134.81]

* What you're doing and what's happening:
I'm not behind proxy
I'm using portable hardisk
I'm using Kaspersky Free
I did test using my home and work connection (both not behind proxy)
I did disable AV (both windows and kaspersky) and Firewall
I did run as administrator
```
curl -i http://archive.ubuntu.com -v
* Rebuilt URL to: http://archive.ubuntu.com/
*   Trying 91.189.88.161...
* Connected to archive.ubuntu.com (91.189.88.161) port 80 (#0)
> GET / HTTP/1.1
> Host: archive.ubuntu.com
> User-Agent: curl/7.47.0
> Accept: */*
>
* Empty reply from server
* Connection #0 to host archive.ubuntu.com left intact
curl: (52) Empty reply from server
```

* What's wrong / what should be happening instead: Response from server isn't empty

* Strace of the failing command, if applicable:  (If `some_command` is failing, then run `strace -o some_command.strace -f some_command some_args`, and link the contents of `some_command.strace` in a [gist](https://gist.github.com/) here)

See [our contributing instructions](https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md) for assistance.

**Comment by rafaeldamasceno** [comment]
I'm also having trouble with connection using Kaspersky Free 19, but it's just (?) HTTP that's not working, so can't `apt-get update` or `curl`.

DNS is working fine, ping works as expected.

I have tried disabling the AV, but that doesn't seem to help.

This seems related to #475, but it's not the same as connection is working.

The closest related issue is #1720, but that involved IPv6 issues and eventually a KES patch, which is not available to Kaspersky Free users.

**Comment by therealkenc** [comment]
> I try to uninstall KF and it's working

That's the lesson. When you disable antivirus software, the too-clever-by-half low level system modifications they make remain in place. If those mods are what is tripping up WSL's low-level assumptions, then badness results whether the antivirus software is "running" or not.

**Comment by rafaeldamasceno** [comment]
@therealkenc Yes, I realized that removing Kaspersky altogether would probably work. However, removing the antivirus is not a good solution nor an option, sometimes. I'd still like to get the incompatibility solved and, following this thought, what do you feel is the best option? Should I just go into #475 and report my issue? Should I hijack this issue? Should I create another one? Or do you feel this problem might be too much on the side of Kaspersky that I should be bothering them instead?

**Comment by therealkenc** [comment]
>   Or do you feel this problem might be too much on the side of Kaspersky that I should be bothering them instead?

It isn't really a "side" thing. The MSFT devs aren't privy to what 3rd-party antivirus vendors are doing under the hood. I am going to venture that you are not the first person to have encountered the problem and that Kaspersky has already been notified, but it doesn't hurt I suppose.

**Comment by vlad-s** [comment]
Sorry to interrupt, the only fix I found for this issue is by removing the port monitoring in Kaspersky Free. This is done via KF - Settings - Network Settings - check "Monitor selected ports only" - Select ports - remove everything. After clearing the port monitoring list, networking in WSL worked without issues.

@rafaeldamasceno & @wicaksono, I know it's ugly, but at least there's no need to uninstall or disable the AV.

![image](https://user-images.githubusercontent.com/7457966/41648334-43f3c35c-7481-11e8-852a-b9f71ce2e1bf.png)
